api/src/components/authorization/controller.ts

```diff
@@ -92,4 +91,9 @@ export const authorize = async (event: I.APIRequest<I.AuthorizeRequestBody>): Pr
         console.log(err);
         return response.json(500, { message: 'Unknown server error.' });
     }
-};
\ No newline at end of file
+};
+
+export const verifyToken = async (event: I.AuthenticatedAPIRequest) => {
```

Comment:
this is really verifying the user or even fetching them, the token has already been processed by the time we get here..

Reply:
It's verifying the token and returns a 200 + decoded info it's valid. It's not fetching the user as that would mean to grab it from DB, and that would be too slow for what we need from this endpoint. The decoded info returned is not 100% necessary. 

On the client side, we only care about that 200 response (atm). But on the server side, I thought we could use this endpoint to verify the token + get some basic user info from it like user ID if we need to fetch some data based on that, instead of decoding it server side without knowing 100% if it's valid or not (only the backend can tell that).

See the example below:
<img width="872" alt="image" src="https://user-images.githubusercontent.com/48569671/192339994-f91bb4d7-fdf0-4ae4-a596-ba5dc86c81ac.png">

We could call verifyToken endpoint from "guardPrivateRoute" helper and get the user ID from there if we need it, or to check if the email is verified, before rendering the page.

Reply:
all sounds good by my point is the method name suggests the method is verifying the Token when it doesn't, it doesnt verify anything it simply returns the decoded token, perhaps call it fetchDecodedTokenUser or something

Reply:
Changed verifyToken to getDecodedUserToken.